core/Network/TLS/X509.hs

```diff
@@ -14,21 +14,28 @@ module Network.TLS.X509
     , getCertificate
     , isNullCertificateChain
     , getCertificateChainLeaf
+    , getSystemCertificateStore
     , CertificateRejectReason(..)
     , CertificateUsage(..)
     , CertificateStore
     , ValidationCache
     , exceptionValidationCache
+    , makeCertificateStore
     , validateDefault
     , FailedReason
+    , readCertificateStore
+    , readCertificates
```

Comment:
Why `readCertificates` instead of:

* `readCertificateStore`: it reads a whole directory, which probably contains file extraneous to the cert chain
* `readSignedObject` it doesn't do error-handling for IO or other runtime errors, unlike the functions exported from `Data.X509.CertificateStore` (I limited the use of `Data.X509.File` to `readKeyFile`, which is the only function that doesn't have an equivalent in `Data.X509.CertificateStore`)